net/pfSense-pkg-haproxy/files/usr/local/pkg/haproxy/haproxy.inc

```diff
@@ -2079,22 +2092,27 @@ function load_ipfw_rules() {
 		$transparent_interfaces[$interface] = 1;
 	}
 	
-	if (haproxy_utils::$pf_version < 2.2) {
-		// pfSense 2.1 FreeBSD 8.3
-		mwexec("/usr/local/sbin/ipfw_context -a $ipfw_zone_haproxy", true);
-		
-		foreach($transparent_interfaces as $transparent_if => $value) {
-			mwexec("/usr/local/sbin/ipfw_context -a $ipfw_zone_haproxy -n $transparent_if", true);
+	if (haproxy_utils::$pf_version < 2.4) {
+		$rulenum = 64000; // why that high? captiveportal.inc also does it...
+		if (haproxy_utils::$pf_version < 2.2) {
+			// pfSense 2.1 FreeBSD 8.3
+			mwexec("/usr/local/sbin/ipfw_context -a $ipfw_zone_haproxy", true);
+
+			foreach($transparent_interfaces as $transparent_if => $value) {
+				mwexec("/usr/local/sbin/ipfw_context -a $ipfw_zone_haproxy -n $transparent_if", true);
+			}
+		} else {
+			// pfSense 2.2 FreeBSD 10
+			mwexec("/sbin/ipfw zone $ipfw_zone_haproxy create", true);
+			foreach($transparent_interfaces as $transparent_if => $value) {
+				mwexec("/sbin/ipfw zone $ipfw_zone_haproxy madd $transparent_if", true);
+			}
 		}
 	} else {
-		// pfSense 2.2 FreeBSD 10
-		mwexec("/sbin/ipfw zone $ipfw_zone_haproxy create", true);
-		foreach($transparent_interfaces as $transparent_if => $value) {
-			mwexec("/sbin/ipfw zone $ipfw_zone_haproxy madd $transparent_if", true);
-		}
+		// pfSense 2.4, captive portal rules start at 1000
+		$rulenum = 10;
```

Comment:
10 is right i think for haproxy, as 'zones' no longer separate stuff in ipfw contexts, and haproxy needs to catch reply traffic it shouldn't use the same rule numbers as CP which does afaik start with rule 1000, putting the rules with for haproxy with a lower number seems to work properly.